.github/workflows/deploy_jupyterbook.yml

```diff
@@ -0,0 +1,51 @@
+name: deploy-book
+
+# Only run this when the main branch changes
+on:
+  push:
+    branches:
+    - main
+
+  pull_request:
+    branches:
+    - main
+
+# This job installs dependencies, build the book, and pushes it to `gh-pages`
+jobs:
+  deploy-book:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    # Install dependencies
+    - name: Set up Python 3.10.12
+      uses: actions/setup-python@v1
+      with:
+        python-version: 3.10.12
+
+    - name: Install dependencies
+      run: |
+        cd pytket-examples-book
+        python -m pip install --upgrade pip wheel
+        python -m pip install -r example_requirements.txt
+        python -m pip install jupyter-book
+
+    # Build the book
+    - name: Build the book
+      run: |
+        cd pytket-examples-book
+        jupyter-book build -W .
```

Comment:
I've added the -W flag so the jupyterbook build will fail for warnings.

 Jupyterbook's default behaviour is to give a warning when a cell fails to execute and show the error in the html. This -W flag promotes the warnings to errors so that the build will always fail if a cell fails to excute.